### ISSUE TITLE
script/setup failing at fetching lucky_flow

### ISSUE THREAD

**Issue by backward-crazy-mage-puppy-36** [luckyframework/lucky#1484]
After installing lucky and initialising a test application, when running script/setup it fails at fetching lucky_flow
github repository.
Error: Failed to clone https://github.com/luckyframework/lucky_flow.git

MacOS Catilina 10.15.7

**Comment by backward-crazy-mage-puppy-36** [comment]
So script/setup now fetches lucky_flow but is failing at
Postinstall script/precompile_tasks
with the error

Postinstall make bin && make run_file
Failed make bin && make run_file:
shards build
Dependencies are satisfied
Building: ameba
Error target ameba failed to compile:
Undefined symbols for architecture x86_64:
  "_iconv", referenced from:
      _*Crystal::Iconv#convert<Pointer(Pointer(UInt8)), Pointer(UInt64), Pointer(Pointer(UInt8)), Pointer(UInt64)>:UInt64 in C-rystal5858I-conv.o
  "_iconv_close", referenced from:
      _*Crystal::Iconv#close:Nil in C-rystal5858I-conv.o
  "_iconv_open", referenced from:
      _*Crystal::Iconv#initialize<String, String, (Symbol | Nil)>:Nil in C-rystal5858I-conv.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: execution of command failed with code: 1: `cc "${@}" -o /Users/bcmp36/code/crystal/lucky/testing/lib/ameba/bin/ameba  -rdynamic -L/usr/local/lib -L/opt/local/lib -lyaml -lpcre -lgc -lpthread /usr/local/Cellar/crystal/1.0.0/src/ext/libcrystal.a -L/opt/local/lib -levent -liconv -ldl`

make: *** [bin/ameba] Error 1

**Comment by jwoertink** [comment]
That's a strange error... So it looks like Crystal 1.0.0 on macOS 10.15.7... I'm assuming you're running Lucky 0.27.0 here... It seems like it's Ameba specific, but I'm not sure. Maybe try clearing out `~/.cache/crystal`, and delete `shard.lock`, and `lib/`, then run it all again?

@veelenga have you seen anything like this before?

**Comment by veelenga** [comment]
Refs:

* https://github.com/crystal-ameba/ameba/issues/232
* https://github.com/crystal-lang/crystal/issues/10653

**Comment by jwoertink** [comment]
oh, boom! Thanks for the update 👍

**Comment by veelenga** [comment]
As a workaround, the post-install can be skipped: `shards install --skip-postinstall`. Not sure if Lucky's scripts allow passing this flag.

**Comment by jwoertink** [comment]
Yeah, that can be done on this line https://github.com/luckyframework/lucky_cli/blob/master/src/web_app_skeleton/script/setup.ecr#L25 

Thanks for the suggestion!

**Comment by backward-crazy-mage-puppy-36** [comment]
> That's a strange error... So it looks like Crystal 1.0.0 on macOS 10.15.7... I'm assuming you're running Lucky 0.27.0 here... It seems like it's Ameba specific, but I'm not sure. Maybe try clearing out `~/.cache/crystal`, and delete `shard.lock`, and `lib/`, then run it all again?
> 
> @veelenga have you seen anything like this before?

I tried clearing out `~/.cache/crystal`, and deleting `shard.lock`, and `lib/` but i am running into problems fetching lucky_flow github repository again.
Error: Failed to clone https://github.com/luckyframework/lucky_flow.git
